src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -2222,5 +2236,16 @@ internal static void UpdateTypeChecking(this FileContentManager file, Compilatio
                 compilation.ExitWriteLock();
             }
         }
+
+        private static void CheckForGlobalCycleChange(FileContentManager file, List<Diagnostic> diagnostics)
+        {
+            var numRemovedCycleDiags = file.CurrentSemanticDiagnostics().Count(m => m.Code == ErrorCode.InvalidCyclicTypeParameterResolution.Code())
+                - diagnostics.Count(m => m.Code == ErrorCode.InvalidCyclicTypeParameterResolution.Code());
+
+            if (numRemovedCycleDiags != 0 || diagnostics.Any(x => x.Source != file.FileName.Value))
+            {
+                file.TriggerGlobalTypeChecking();
+            }
+        }
```

Comment:
I noticed a minor bug. When I have a cycle across two files:

File1.qs
```qsharp
namespace Example {
    operation Foo1<'A>() : Unit {
        Bar<'A>();
    }

    operation Bar<'A>() : Unit {
        Foo2<'A[]>();
    }
}
```

File2.qs
```qsharp
namespace Example {
    operation Foo2<'A>() : Unit {
        Bar<'A>();
    }
}
```

Then saving either of the files (e.g. with Ctrl+S) will cause a red squiggly to flicker in for a split second before disappearing. In the first file, the squiggly is under the call to `Bar`. In the second file, the squiggly is at the last character in the file.

Reply:
I've change the code to always trigger the global type checking. Can you try again?

Reply:
It seems like it still happens.

Reply:
The issue seems to have been fixed by not keeping errors for callables that are not fully loaded into what the editor considers.

Reply:
Thanks, it seems like it's fixed now.